### PR TITLE
added dto and web controller

### PR DESCRIPTION
**Pull request by Trayvor**
Added: dtos, mapper, web controller, custom exceptions, reworked repository

### CODE REVIEW

File: src/main/java/org/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/books")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

add `server.servlet.context-path=/api` to application.properties

File: src/main/java/org/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -10,36 +10,21 @@
 import lombok.Data;
 
 @Entity
-@Table(name = "books")
 @Data
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java]: Don't use `@Data` for entities, read which annotations it includes and use necessary

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -1,26 +1,41 @@
...
+    private final BookMapper bookMapper;
 
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book model = bookMapper.toModel(requestDto);
+        bookRepository.save(model);
+        BookDto dto = bookMapper.toDto(model);
+        return dto;
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
        return bookMapper.toDto(model);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
good job. see some comments

File: src/main/java/org/example/bookstore/model/Book.java [hunk 4]
```diff
@@ -5,41 +5,28 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 
 @Entity
-@Table(name = "books")
-@Data
+@Getter
+@Setter
+@NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java]:
```suggestion
```

File: src/main/java/org/example/bookstore/model/Book.java [hunk 5]
```diff
@@ -5,41 +5,28 @@
...
-import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 
 @Entity
-@Table(name = "books")
-@Data
+@Getter
+@Setter
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java:15]:
```suggestion
@Setter
@Table(name = "books")
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**